apps/aecore/include/txs.hrl

```diff
@@ -0,0 +1,77 @@
+-type(pubkey() :: binary()).
+
+-record(coinbase_tx, {account = <<>> :: pubkey(),
+                      nonce = 0      :: non_neg_integer()}).
+
+-record(spend_tx, {from = <<>> :: pubkey(),
+                   to = <<>>   :: pubkey(),
+                   amount = 0  :: non_neg_integer(),
+                   fee = 0     :: non_neg_integer(),
+                   nonce = 0   :: non_neg_integer()}).
+
+
+%% Channel-related transactions
+
+-record(channel_new_tx, {id = 0          :: non_neg_integer(),
+                         account1 = <<>> :: pubkey(),
+                         account2 = <<>> :: pubkey(),
+                         amount1 = 0     :: non_neg_integer(),
+                         amount2 = 0     :: non_neg_integer(),
+                         fee = 0         :: non_neg_integer(),
+                         nonce = 0       :: non_neg_integer(),
+                         delay = 0       :: non_neg_integer()}).
+
+-record(channel_grow_tx, {id = 0          :: non_neg_integer(),
+                          account1 = <<>> :: pubkey(),
+                          account2 = <<>> :: pubkey(),
+                          amount1 = 0     :: non_neg_integer(),
+                          amount2 = 0     :: non_neg_integer(),
+                          fee = 0         :: non_neg_integer(),
+                          nonce = 0       :: non_neg_integer(),
+                          delay = 0       :: non_neg_integer()}).
+
+-record(channel_team_close_tx, {id = 0          :: non_neg_integer(),
+                                account1 = <<>> :: pubkey(),
+                                account2 = <<>> :: pubkey(),
+                                amount = 0      :: pubkey(),
+                                fee = 0         :: non_neg_integer(),
+                                nonce = 0       :: non_neg_integer()}).
+
+-record(channel_solo_close_tx, {id = 0         :: non_neg_integer(),
```

Comment:
self close?

Reply:
I didn't comment on the naming because some of the transactions may go eventually. I do think that having both close and solo close is superfluous as the system should be able to recognize that you are closing the channel unilaterally. Besides, it makes for less conflicts to resolve, e.g. when one side did a close and the other one a solo close.

Reply:
Well.. do we want to evaluate these transactions now, or leave it as it is, and massage when channels implementation takes place?